### ISSUE TITLE
reportIncompatibleMethodOverride not working for protected methods, i.e. whose names start with underscore

### ISSUE THREAD

**Issue by vishwa-raj** [microsoft/pylance-release#843]
<h3> Environment data </h3>

-   Language Server version: 2021.1.1
-   OS and version: win32 x64
-   Python version (and distribution if applicable, e.g. Anaconda): Python 3.7 Anaconda
-   python.analysis.typeCheckingMode: strict
-   Override settings:
```
    "python.analysis.diagnosticSeverityOverrides": {
        "reportMissingTypeArgument": "none",
        "reportUnusedImport": "none",
        "reportUnusedVariable": "none" }
```
<h3> Expected behaviour </h3>

In the following example, I was expecting both, `method()` and `_protected_method()` in `Child` class, to be flagged for `reportIncompatibleMethodOverride` :

![image](https://user-images.githubusercontent.com/22815753/104824637-87f1f400-5879-11eb-8b91-37b5e5bdc714.png)

<h3> Actual behaviour </h3>

But only `method()` is being flagged:

![image](https://user-images.githubusercontent.com/22815753/104824656-ac4dd080-5879-11eb-9d1f-cf51c143c377.png)

**Comment by erictraut** [comment]
Good catch. When I originally wrote this logic, I explicitly excluded methods that start with a single underscore, but I agree that they should not be excluded from this check.

This will be fixed in the next release.

**Comment by vishwa-raj** [comment]
Thanks @erictraut

**Comment by erictraut** [comment]
Fixed in this PR: https://github.com/microsoft/pyright/commit/3c1576ba2941775bff4cf59840f68fe9bf32fa9e

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.1.2, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202112-20-january-2021
